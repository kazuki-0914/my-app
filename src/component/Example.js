import React, { Component } from 'react';

export default class Example extends Component {
  constructor() {
    super();

    this._onClick = this._onClick.bind(this);

    this.state = {
      count: 0
    }
  }

  _onClick() {
    this.setState( { count: this.state.count + 1 } );
  }

  render() {
    return (
      <div>
        <p>{this.state.count}</p>
        <button onClick={this._onClick} >Click</button>
      </div>
      )
  }
}