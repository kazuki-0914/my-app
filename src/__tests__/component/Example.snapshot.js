'use strict';

import React from 'react';
import Example from '../../component/Example';
import renderer from 'react-test-renderer';

describe('Example', () => {

  it('レンダリングができること', () => {
    const tree = renderer.create(
      <Example />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('onClick', () => {
    const component = renderer.create(
      <Example />
    )
    // 1回目クリック
    component.getInstance()._onClick();
    expect(component.toJSON()).toMatchSnapshot();
    // 2回目クリック
    component.getInstance()._onClick();
    expect(component.toJSON()).toMatchSnapshot();
  });

});