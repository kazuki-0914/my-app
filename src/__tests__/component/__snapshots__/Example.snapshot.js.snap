// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Example onClick 1`] = `
<div>
  <p>
    1
  </p>
  <button
    onClick={[Function]}
  >
    Click
  </button>
</div>
`;

exports[`Example onClick 2`] = `
<div>
  <p>
    2
  </p>
  <button
    onClick={[Function]}
  >
    Click
  </button>
</div>
`;

exports[`Example レンダリングができること 1`] = `
<div>
  <p>
    0
  </p>
  <button
    onClick={[Function]}
  >
    Click
  </button>
</div>
`;
